pkg/controller/unreachable/unreachable_controller.go

```diff
@@ -145,88 +144,84 @@ func (r *ReconcileRemoteMachineSet) Reconcile(request reconcile.Request) (reconc
 		return reconcile.Result{}, nil
 	}
 
-	// Check if we're due for rechecking cluster's connectivity
-	cond := controllerutils.FindClusterDeploymentCondition(cd.Status.Conditions, hivev1.UnreachableCondition)
-	if cond != nil {
-		if !isTimeForConnectivityCheck(cond) {
-			cdLog.WithFields(log.Fields{
-				"sinceLastTransition": time.Since(cond.LastTransitionTime.Time),
-				"sinceLastProbe":      time.Since(cond.LastProbeTime.Time),
-			}).Debug("skipping unreachable check")
-			return reconcile.Result{}, nil
-		}
+	wasUnreachable, lastCheck := remoteclient.Unreachable(cd)
+	wasPrimaryActive := remoteclient.IsPrimaryURLActive(cd)
+	connectivityCheckDelay := maxUnreachableDuration - time.Since(lastCheck)
+
+	if !wasUnreachable && wasPrimaryActive && connectivityCheckDelay > 0*time.Second {
```

Comment:
Comments needed on this if. Wondering why wasPrimaryActive is called out here, and connectivityCheckDelay as well.